akka-projection-grpc-tests/src/it/scala/akka/projection/grpc/replication/EdgeReplicationMultiIntegrationSpec.scala

```diff
@@ -0,0 +1,297 @@
+
+    // this is the tricky part, binding both these into listening services
+    (replication1, replication2)
+  }
+
+  def startEdgeReplica(): (EdgeReplication[LWWHelloWorld.Command], EdgeReplication[LWWHelloWorld.Command]) = {
+    def replicationSettings(entityTypeKey: EntityTypeKey[LWWHelloWorld.Command]) = {
+      ReplicationSettings[LWWHelloWorld.Command](
+        entityTypeKey.name,
+        EdgeReplicaC,
+        EventProducerSettings(edgeSystem),
+        Set(cloudReplica),
+        10.seconds,
+        1,
+        R2dbcReplication())
+    }
+
+    val replication1 =
+      Replication.grpcEdgeReplication(replicationSettings(LWWHelloWorld.EntityType))(LWWHelloWorld.apply)(edgeSystem)
+    val replication2 =
+      Replication.grpcEdgeReplication(replicationSettings(LWWHelloWorld2.EntityType))(LWWHelloWorld.apply)(edgeSystem)
+
+    // this one is straightforward, no gRPC services bound, only locally running sharding and event push-projections
+    (replication1, replication2)
+  }
+
+  def assertGreeting(entityTypeKey: EntityTypeKey[LWWHelloWorld.Command], entityId: String, expected: String): Unit = {
+    testKitsPerDc.values.foreach { testKit =>
+      withClue(s"on ${testKit.system.name}") {
+        val probe = testKit.createTestProbe()
+        withClue(s"for entity id $entityId") {
+          val entityRef = ClusterSharding(testKit.system)
+            .entityRefFor(entityTypeKey, entityId)
+
+          probe.awaitAssert({
+            entityRef
+              .ask(LWWHelloWorld.Get.apply)
+              .futureValue should ===(expected)
+          }, 10.seconds)
+        }
+      }
+    }
+  }
+
+  "Replication over gRPC" should {
+    "form one node clusters" in {
+      testKitsPerDc.values.foreach { testKit =>
+        val cluster = Cluster(testKit.system)
+        cluster.manager ! Join(cluster.selfMember.address)
+        testKit.createTestProbe().awaitAssert {
+          cluster.selfMember.status should ===(MemberStatus.Up)
+        }
+      }
+    }
+
+    "start replicas" in {
+      logger
+        .infoN(
+          "Starting replica [{}], system [{}] on port [{}]",
+          cloudReplica.replicaId,
+          system.name,
+          cloudReplica.grpcClientSettings.defaultPort)
+
+      val (replication1, replication2) = startCloudReplica()
+      val grpcPort = cloudReplica.grpcClientSettings.defaultPort
+
+      // same Akka Projection gRPC for both entities for the edge pulling events
+      val consumeHandler =
+        EventProducer.grpcServiceHandler(Set(replication1.eventProducerSource, replication2.eventProducerSource))(
+          system)
+
+      // same Akka Projection gRPC service for both entities for the edge pushing events
+      // FIXME annoying that it is an option, could we type our way around it somehow?
+      val produceHandler = EventProducerPushDestination.grpcServiceHandler(
+        Set(replication1.eventProducerPushDestination, replication2.eventProducerPushDestination).flatten)(system)
```

Comment:
This is what was not possible before - combining multiple EventProducerPushDestinations for RES entities into one gRPC service instance/handler.